hummingbot/notifier/telegram_notifier.py

```diff
@@ -0,0 +1,154 @@
+        self._updater = Updater(token=token, workers=0)
+        self._hb = hb
+        self._ev_loop = asyncio.get_event_loop()
+
+        # Register command handler and start telegram message polling
+        handles = [MessageHandler(Filters.text, self.handler)]
+        for handle in handles:
+            self._updater.dispatcher.add_handler(handle)
+
+    def start(self):
+        if not self._started:
+            self._started = True
+            self._updater.start_polling(
+                clean=True,
+                bootstrap_retries=-1,
+                timeout=30,
+                read_latency=60,
+            )
+            self.logger().info("Telegram is listening...")
+
+    def stop(self) -> None:
+        if self._started or self._updater.running:
+            self._updater.stop()
+
+    @authorized_only
+    def handler(self, bot: Bot, update: Update) -> None:
+        try:
+            input_text = update.message.text.strip()
+            output = f"\n[Telegram Input] {input_text}"
+            self._hb.app.log(output)
+
+            # if the command does starts with any disabled commands
+            if any([input_text.startswith(dc) for dc in DISABLED_COMMANDS]):
+                self.send_msg(f"Command {input_text} is disabled from telegram", bot=bot)
+            else:
+                self._hb._handle_command(input_text)
+        except Exception as e:
+            self.send_msg(str(e), bot=bot)
+
+    @staticmethod
+    def _divide_chunks(arr: List[Any], n: int = 5):
+        """ Break a list into chunks of size N """
+        for i in range(0, len(arr), n):
+            yield arr[i:i + n]
+
+    def send_msg(self, msg: str, bot: Bot = None) -> None:
+        asyncio.ensure_future(self.send_msg_async(msg, bot), loop=self._ev_loop)
+
+    async def send_msg_async(self, msg: str, bot: Bot = None) -> None:
+        """
+        Send given markdown message
+        """
+        bot = bot or self._updater.bot
+
+        # command options that show up on user's screen
+        approved_commands = [c for c in self._hb.parser.commands if c not in DISABLED_COMMANDS]
+        keyboard = self._divide_chunks(approved_commands)
+        reply_markup = ReplyKeyboardMarkup(keyboard)
+
+        # wrapping text in ``` to prevent formatting issues
+        formatted_msg = f'```\n{msg}\n```'
+
+        try:
+            try:
+                bot.send_message(
+                    self._chat_id,
+                    text=formatted_msg,
+                    parse_mode=ParseMode.MARKDOWN,
+                    reply_markup=reply_markup
+                )
+            except NetworkError as network_err:
+                # Sometimes the telegram server resets the current connection,
+                # if this is the case we send the message again.
+                self.logger().warning('Telegram NetworkError: %s! Trying one more time.', network_err.message)
+                bot.send_message(
```

Comment:
Update this to `scheduler.call_async()` as well. This time include a `app_warning_msg` since it may be that the user has used an incorrect token or is having network connectivity issue.